book/src/dev/rfcs/0005-treestate.md

```diff
@@ -0,0 +1,194 @@
+# Treestate
+
+- Feature Name: treestate
+- Start Date: 2020-08-31
+- Design PR: [ZcashFoundation/zebra#983](https://github.com/ZcashFoundation/zebra/issues/983)
+- Zebra Issue: [ZcashFoundation/zebra#958](https://github.com/ZcashFoundation/zebra/issues/958)
+
+# Summary
+[summary]: #summary
+
+To validate blocks involving shielded transactions, we have to check the
+computed treestate from the included transactions against the block header
+metadata (for Sapling) or previously finalized state (for Sprout). This document
+describes how we compute and manage that data, assuming a finalized state
+service as described in the [State Updates RFC](https://zebra.zfnd.org/dev/rfcs/0005-state-updates.md).
+
+
+# Motivation
+[motivation]: #motivation
+
+Block validation requires checking that the treestate of the block (consisting
+of the note commitment tree and nullifier set) is consistent with the metadata
+we have in the block header (the root of the note commitment tree) or previously
+finalized state (for Sprout).
+
+
+# Definitions
+[definitions]: #definitions
```

Comment:
It might be helpful to split the definitions into Sprout and Sapling sections, and a common section.

Reply:
Added as a TODO in this section:
https://github.com/ZcashFoundation/zebra/pull/983/commits/e9e5afa78b776e22cd4e7abc186d97626a27f53c